src/components/editor/metaform-editor-right-drawer.tsx

```diff
@@ -220,14 +263,18 @@ const MetaformEditorRightDrawer: React.FC<Props> = ({
               <MenuItem sx={{ color: "gray" }}>{ strings.draftEditorScreen.editor.visibility.selectField }</MenuItem>
               { pendingForm.sections!.map(section => section.fields!.map((field, index) => {
                 const constructedKey = `${field.title}-${index}`;
-                const currentField = pendingForm.sections![sectionIndex].fields![fieldIndex];
-                if (currentField.name === field.name) {
+                
+                if (field.type === "radio" || field.type === "checklist" || field.type === "boolean") {
+                  if (field.name !== selectedComponent) {
+                    return (
+                      <MenuItem value={ field.name } key={ constructedKey }>
+                        { field.title }
+                      </MenuItem>
+                    );
+                  }
                   return null;
-                } return (
-                  <MenuItem value={ field.name } key={ constructedKey }>
-                    { field.title }
-                  </MenuItem>
-                );
+                }
+                return null;
```

Comment:
Aren't these _return null;_'s redundant?

Reply:
Other return was useless but Array.prototype.map() expects a value to be returned at the end of arrow function